src/WebJobs.Script/Host/WorkerFunctionMetadataProvider.cs

```diff
@@ -89,7 +92,7 @@ public async Task<FunctionMetadataResult> GetFunctionMetadataAsync(IEnumerable<R
                     if (IsJobHostStarting())
                     {
                         _logger.LogDebug("JobHost is starting with state '{State}'. Initializing worker channel.", _scriptHostManager.State);
-                        await _channelManager.InitializeChannelAsync(workerConfigs, _workerRuntime);
+                        await _channelManager.InitializeChannelAsync(workerConfigs, _workerRuntime, _applicationLifetime.ApplicationStopping);
```

Comment:
So my idea was to pass the `_applicationLifetime.ApplicationStopping` CT to `StartWorkerProcessAsync` -> PendingItem. (it doesn't have to come through `WorkerFunctionMetadataProvider` so let me know if you have better ideas)

The problem I am seeing with this is that when I cancel (exit the host) we don't see CT triggered until _after_ the timeout period of `PendingItem` has run its course. We are blocking in a weird way and we need to get around that block for the CT to flow properly.

cc: @jviau @fabiocav